dagshub/data_engine/model/query_result.py

```diff
@@ -401,6 +404,78 @@ def download_binary_columns(
+
+        Args:
+            repo: repository to extract the model from
+            name: name of the model in the mlflow registry
+            version: (optional, default: 'latest') version of the model in the mlflow registry
+            pre_hook: (optional, default: identity function) function that runs before datapoint is sent to the model
+            post_hook: (optional, default: identity function) function that converts mlflow model output
+            to the desired format
+            batch_size: (optional, default: 1) function that sets batch_size
+            log_to_field: (optional, default: 'prediction') write prediction results to metadata logged in data engine.
+            If None, just returns predictions.
+            return_predictions: (optional, default: True) returns prediction result
+            (in addition to logging to a field, iff that parameter is set)
+        """
+        if not log_to_field and not return_predictions:
+            raise ValueError("Either `log_to_field` or `return_predictions` must be set.")
+
+        # to support depedency-free dataloading, `Batcher` is a barebones dataloader that sets up batched inference
+        class Batcher:
+            def __init__(self, dset, batch_size):
+                self.dset = dset
+                self.batch_size = batch_size
+
+            def __iter__(self):
+                self.curr_idx = 0
+                return self
+
+            def __next__(self):
+                self.curr_idx += self.batch_size
+                return [self.dset[idx] for idx in range(self.curr_idx - self.batch_size, self.curr_idx)]
+
+        init(*repo.split("/")[::-1])
+        model = mlflow.pyfunc.load_model(f"models:/{name}/{version}")
+        init(*self.datasource.repo.split("/")[::-1])
+
+        dset = DagsHubDataset(self, tensorizers=[lambda x: x])
+
+        predictions = []
+        progress = get_rich_progress(rich.progress.MofNCompleteColumn())
+        task = progress.add_task("Running inference...", total=len(dset))
+        with progress:
+            for idx, local_paths in enumerate(
+                Batcher(dset, batch_size) if batch_size != 1 else dset
+            ):  # encapsulates dataset with batcher if necessary and iterates over it
+                for prediction, remote_path in zip(
+                    post_hook(model.predict(pre_hook(local_paths))),
+                    [result.path for result in self[idx * batch_size: (idx + 1) * batch_size]],
+                ):
+                    predictions.append({remote_path: prediction})
+                progress.update(task, advance=batch_size, refresh=True)
+
+        if not log_to_field:
+            return predictions
+        with self.datasource.metadata_context() as ctx:
+            for remote_path in predictions:
+                ctx.update_metadata(remote_path, {log_to_field: predictions[remote_path]})
+        return predictions if return_predictions else True
```

Comment:
If you delete the `return_predicitons` then this would be:

```suggestion
        if log_to_field:
            with self.datasource.metadata_context() as ctx:
                for remote_path in predictions:
                    ctx.update_metadata(remote_path, {log_to_field: predictions[remote_path]})
        return predictions
```